### PR TITLE
Add figurl backend for plot_timeseries

### PR DESCRIPTION
**Pull request by alejoe91**

### CODE REVIEW

**Comment by alejoe91** [comment]
@samuelgarcia ready to test: depends on this PR https://github.com/scratchrealm/figurl-tiled-image/pull/4

File: spikeinterface/widgets/figurl/timeseries.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        else:
+            color_ranges = 200
+        if not isinstance(color_ranges, list):
+            color_ranges = [color_ranges] * len(data_plot["layer_keys"])
+            
+        if "tile_size" in backend_kwargs:
+            tile_size = backend_kwargs["tile_size"]
+        else:
+            tile_size = 512
+
```
Comment (samuelgarcia) [inline spikeinterface/widgets/figurl/timeseries.py:31]:
```suggestion
            tile_size = backend_kwargs.get("tile_size", 512)
```
I would do that but more generaly I would create a default dict that would be copied and update with backend_kwargs no ?

**Comment by alejoe91** [comment]
@jsiegle I changed the `color_range` of the `array_to_image` to a `clim`, so it can be used in the same way by both the matplotlib, ipywidgets, and figurl backends. 
Now the syntax to use different color ranges would be:

```
si.plot_timeseries({"filt": rec_f, "cmr": rec_cmr}, 
                   time_range=[10, 10.4], 
                   order_channel_by_depth=True,
                   backend="figurl",
                   tile_size=1024,
                   clim=[[-100, 100], [-500, 500]],
                   seconds_per_row=0.2)
```
which produces this: https://figurl.org/f?v=gs://figurl/figurl-tiled-image-2&d=sha1://1396953737bd4ba79e99784209ec10cf981b3f33&label=SpikeInterface%20-%20%5B%27filt%27%2C%20%27cmr%27%5D

@samuelgarcia ready to review! I also changed allowed the ipywidgets backend to have differnt clims for different layers

File: spikeinterface/widgets/ipywidgets/timeseries.py [hunk 2]
```diff
@@ -92,6 +93,15 @@ def __init__(self, data_plot, mpl_plotter, ax, time_slider, seg_selector, win_si
...
         self.layer_selector = layer_selector
         
+        clim = data_plot["clim"]
+        if clim is not None:
+            if np.array(clim).ndim == len(data_plot["layer_keys"]):
+                self.clims = clim
+            else:
+                self.clims = [clim] * len(data_plot["layer_keys"])
+        else:
+            self.clims = None
```
Comment (samuelgarcia) [inline spikeinterface/widgets/ipywidgets/timeseries.py:103]: A dict woulmd be easier here no ?

File: spikeinterface/widgets/ipywidgets/timeseries.py [hunk 3]
```diff
@@ -127,13 +137,19 @@ def __call__(self, change):
             # plot all layer
             layer_keys = self.data_plot['layer_keys']
             recordings = self.recordings
+            clim = None
         elif mode =='map':
             layer_keys = [selected_layer]
             recordings = {selected_layer: self.recordings[selected_layer]}
+            if self.clims is not None:
+                clim = self.clims[list(self.recordings.keys()).index(selected_layer)]
```
Comment (samuelgarcia) [inline spikeinterface/widgets/ipywidgets/timeseries.py:145]: dict would be more easy to manipualte

File: spikeinterface/widgets/timeseries.py [hunk 4]
```diff
@@ -37,10 +36,15 @@ class TimeseriesWidget(BaseWidget):
...
         The color used to draw the traces.
-    clim: None or tupple
-        When mode='map' this control color lims
+    clim: None, tuple, or list of tuples
+        When mode='map' this control color lims. 
+        If list of tuples, each entry is the color lims for one layer.
     with_colorbar: bool default True
         When mode='map' add colorbar
+    tile_size: int
+        For figurl backend, the size of each tile in the redered image
```
Comment (samuelgarcia) [inline spikeinterface/widgets/timeseries.py:45]: This should not be int but something in second or millisecond and then internally converted to tile_size

Comment (samuelgarcia) [reply spikeinterface/widgets/timeseries.py:45]: oups no sorry I mixed witth seconds_per_row

File: spikeinterface/widgets/ipywidgets/base_ipywidgets.py [hunk 5]
```diff
@@ -7,5 +7,13 @@
 
 class IpywidgetsPlotter(BackendPlotter):
     backend = 'ipywidgets'
+    backend_kwargs = {
```
Comment (samuelgarcia) [inline spikeinterface/widgets/ipywidgets/base_ipywidgets.py:10]: description_backend_kwargs

File: spikeinterface/widgets/timeseries.py [hunk 6]
```diff
@@ -37,10 +36,15 @@ class TimeseriesWidget(BaseWidget):
         If True groups are plotted with different colors
     color:   str default: None
         The color used to draw the traces.
-    clim: None or tupple
-        When mode='map' this control color lims
+    clim: None, tuple, or list of tuples
+        When mode='map' this control color lims. 
+        If list of tuples, each entry is the color lims for one layer.
```
Comment (samuelgarcia) [inline spikeinterface/widgets/timeseries.py:41]: If dict (or list)

File: spikeinterface/widgets/figurl/timeseries.py [hunk 7]
```diff
@@ -17,34 +17,17 @@ class TimeseriesPlotter(FigurlPlotter):
...
-            tile_size = data_plot["tile_size"]
-        else:
-            tile_size = 512
-            
-        if "num_timepoints_per_row" in data_plot:
-            num_timepoints_per_row = data_plot["num_timepoints_per_row"]
-        else:
-            num_timepoints_per_row = 6000
+        tile_size = data_plot.get("tile_size", 512)
+        num_timepoints_per_row = data_plot.get("num_timepoints_per_row", 6000)
```
Comment (samuelgarcia) [inline spikeinterface/widgets/figurl/timeseries.py:21]: Why do we need default value security here ?
For me all defaults have to be handle in the base.
All plotters can digest without checks the data_plot dict.

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**